### DISCUSSION TITLE
Is it possible to use custom HTML for home/root page?

### DISCUSSION THREAD

**Discussion by elringus** [vuejs/vitepress#2425] [Q&A] [answered]
In vuepress I was able to just upload `.vuepress/public/index.html` and it was picked instead of default home.

But in vitepress I can't find a way to do this. The custom index is accessible via `mysite.com/index.html` but `mysite.com` throws 404.

First I've tried wrapping the HTML page inside a vue template component and injecting it to `docs/index.md` with `layout: false` frontmatter, but all the JS scripts were not working and the page itself was rendered inside vue app, which is not desired.

I've tried using route rewrites, but they don't seem to support routing to static html pages.

I feel it should be possible to override routing with `enhanceApp` in `theme/index.ts`, but I can't find any examples on how to do this. Here is what I've tried:

```ts
import Theme from "vitepress/theme";
import "./style.css";

export default {
    ...Theme,
    enhanceApp({ app, router, siteData }) {
        router.addRoute({ path: "/", redirect: "/public/index.html" });
    }
};
```

— but it throws "router.addRoute is not a function", even though it should be judging by https://router.vuejs.org/api/interfaces/Router.html#Methods-addRoute.

Would be very grateful for any hints on how this could be achieved. Basically, I just need domain root to redirect to a custom static HTML, while everything else should work by default. I'm aware it's possible with URL rewriting on HTTP server side, but I'm hosting the app on github pages, so that's not possible.

**Comment by brc-dd** [comment]
Move public directory outside the .vitepress directory

**Reply by elringus** [reply]
It's not inside vitepress, it's in `docs/public`.  It was under `.vuepress` when I was using vuepress, I'm aware in vite it's in another location.

**Comment by elringus** [comment, marked as answer]
Solved in https://github.com/vuejs/vitepress/issues/2426
